packages/component-meta/scripts/generateComponentMeta.ts

```diff
@@ -298,25 +313,48 @@ const generateComponentMeta = (): void => {
                   (result[propName] as string[]).push(undefined);
                 }
 
-                let [, variable] = (values.match(/([A-Z_]{5,})/) || []) as [string, string | string[]];
+                const [, variable] = values.match(/([A-Z_]{5,})/) || [];
                 if (variable) {
-                  const variableExportFilePath = getExportFilePath(
-                    source.includes(variable as string) ? source : sourceWithSharedProps,
-                    variable as string,
+                  const variableImportFilePath = getImportFilePath(
+                    source.includes(variable) ? source : sourceWithSharedProps,
+                    variable,
                     tagName
                   );
 
-                  variable = require(variableExportFilePath)[variable as string];
+                  // can be shared utils barrel, cross import from other component or component utils
+                  const variableModule = require(variableImportFilePath);
+                  let variableValues: string[] = variableModule[variable];
+
+                  // check if there is a _DEPRECATED array to import
+                  if (variableModule[`${variable}_DEPRECATED`]) {
```

Comment:
This implies that we have to create a `_DEPRECATED` array even if there is only a single deprecated value? e.G. `export const LINK_TILE_WEIGHTS = [...TILE_WEIGHTS, 'semibold'] as const;` where `semibold` is deprecated is not covered here?

Reply:
This reads the `_DEPRECATED` array if there is one. If there is none, than we can't help it for now.